packages/hardhat-test-helpers/src/chai/matchers/bigNumber.ts

```diff
@@ -0,0 +1,264 @@
+// TODO: consider adding support for the following Chai.Assert properties.
+// Might any of these come "for free" due to supporting other methods? At the
+// very least get them tested.
+//   isAbove
+//   isAtLeast
+//   isBelow
+//   isAtMost
+//   isAt
+//   isNumber
+//   isNotNumber
+//   closeTo
+//   approximately
+
+import { BigNumber as EthersBigNumberType } from "ethers";
+// eslint-disable-next-line import/no-extraneous-dependencies
+import { BigNumber as BigNumberJsType } from "bignumber.js";
+// eslint-disable-next-line import/no-extraneous-dependencies
+import { default as BNType } from "bn.js";
+
+function isBN(n: any) {
+  try {
+    // eslint-disable-next-line import/no-extraneous-dependencies
+    const BN: typeof BNType = require("bn.js");
+    return n instanceof BN && BN.isBN(n);
```

Comment:
I did some experimentation in an external client package, and learned first-hand what you mean here. I dropped the `instanceof` checks in https://github.com/NomicFoundation/hardhat/pull/2428/commits/bd84dd0f930960f6d588141a3f058f508bb87e31. Also in that commit I simplified the conversion of a `bignumber.js` `BigNumber` to `BigInt` because I realized that that `BigNumber`'s `.toString()` argument *defaults to* `10`, so we don't actually need to specify it, so we can collapse the conversion to just one code path, so it actually doesn't matter if `ethers.BigNumber.isBigNumber` returns true for instances of `bignumber.js`.